suite-native/module-send/src/components/CustomFeeBottomSheet.tsx

```diff
@@ -0,0 +1,114 @@
+import { useSelector } from 'react-redux';
+import Animated, {
+    FadeInDown,
+    FadeOutDown,
+    SlideInDown,
+    SlideOutDown,
+    useAnimatedStyle,
+    withTiming,
+} from 'react-native-reanimated';
+
+import { useRoute } from '@react-navigation/native';
+
+import { AlertBox, BottomSheet, Button, HStack, Text, VStack } from '@suite-native/atoms';
+import { Translation } from '@suite-native/intl';
+import { useFormContext } from '@suite-native/forms';
+import { CryptoAmountFormatter, CryptoToFiatAmountFormatter } from '@suite-native/formatters';
+import { AccountsRootState, selectAccountNetworkSymbol } from '@suite-common/wallet-core';
+import { SendStackParamList, SendStackRoutes, StackProps } from '@suite-native/navigation';
+
+import { SendFeesFormValues } from '../sendFeesFormSchema';
+import { CustomFeeInputs } from './CustomFeeInputs';
+import { useCustomFee } from '../hooks/useCustomFee';
+
+type CustomFeeBottomSheetProps = {
+    isVisible: boolean;
+    onClose: () => void;
+};
+
+type RouteProps = StackProps<SendStackParamList, SendStackRoutes.SendAddressReview>['route'];
+
+export const CustomFeeBottomSheet = ({ isVisible, onClose }: CustomFeeBottomSheetProps) => {
+    const route = useRoute<RouteProps>();
+    const { accountKey, tokenContract } = route.params;
+
+    const { feeValue, isFeeLoading, isSubmittable, isErrorBoxVisible } = useCustomFee({
+        accountKey,
+        tokenContract,
+    });
+
+    const networkSymbol = useSelector((state: AccountsRootState) =>
+        selectAccountNetworkSymbol(state, accountKey),
+    );
+
+    const { setValue, handleSubmit } = useFormContext<SendFeesFormValues>();
+
+    const handleSetCustomFee = handleSubmit(() => {
+        setValue('feeLevel', 'custom');
+        onClose();
+    });
+
+    const animatedButtonContainerStyle = useAnimatedStyle(
```

Comment:
The layout linear transition works good for calculating animation of the component wrappen in it. But I decided to use this kind of height transformation, because it also nicely resizes the whole bottomSheet (wrapper of the content) without doing any nasty layout jumps. If you have some better solution to address this issue, I will be happy to fix that.

Reply:
cool didnt know that, thanks for explanation